arrabbiata/src/setup.rs

```diff
@@ -261,9 +268,36 @@ where
                 })
         };
 
-        // FIXME: setup correctly the initial sponge state
-        let sponge_e1: [BigInt; PlonkSpongeConstants::SPONGE_WIDTH] =
-            std::array::from_fn(|_i| BigInt::from(42u64));
+        // The initial transcript state, i.e. all the messages the verifier and
+        // prover know before any computation, are all the commitments to the
+        // selectors for both circuit.
+        // We create an initial sponge state by instantiating a sponge on both
+        // curves and absorb the corresponding commitments.
+        // After that, we squeeze the state of one sponge to absorb in the
+        // other.
+        // Arbitrarily, we pick the sponge over second curve to output a digest,
+        // and we take the state of the sponge over the first curve.
+        let mut sponge_e1 = E1::create_new_sponge();
```

Comment:
Note that we could simply use Blake2b with a concatenated bytes sequence with all the information.